src/mod_mam_muc.erl

```diff
@@ -208,8 +206,9 @@ stop(Host) ->
 %% @doc Handle public MUC-message.
 -spec filter_room_packet(Packet :: packet(), EventData :: list()) -> packet().
 filter_room_packet(Packet, EventData) ->
+    {room_jid, #jid{lserver = LServer}} = lists:keyfind(room_jid, 1, EventData),
     ?DEBUG("Incoming room packet.", []),
-    IsArchivable = mod_mam_muc_params:is_archivable_message(?MODULE, incoming, Packet),
```

Comment:
I can't see now that we are using function that is configuration depended.
I would rather keep mod_mam_muc_params, by making a real mod_mam_muc_params module.
But of course, you can call gen_mod from inside this new module.

mod_mam_utils is supposed to have functions, that would always work the same predictable way, not depending on configuration parameters.

Reply:
> I can't see now that we are using function that is configuration depended.

If not religiously adhering to dependency injection by parameters, *when* can you see whether called function is or is not configuration dependent?

> mod_mam_utils is supposed to have functions, that would always work the same predictable way, not depending on configuration parameters.

Why? I see `mod_mam_utils` as the place for common utility functions of `mod_mam` and `mod_mam_muc`. These new functions are not even the first ones there to call `gen_mod:get_module_opt`.

Reply:
utils - functions that do something.
params - functions that are allowed to override.
I don't like god-modules, we would end-up with another c2s or muc_room.

Reply:
I disagree that this is a way for mod_mam_utils to become a god module, but I agree that a separation into mod_mam_params would be beneficial for code clarity.

Reply:
Done